### PR TITLE
feat(smart-contracts): move Lock proxy to a standalone contract

### PR DESCRIPTION
**Pull request by clemsos**
# Description

This creates a `LockProxy` standalone (flatten) contract. That will make verification process easier across chains.

# Issues

Fixes #
Refs #

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: smart-contracts/contracts/Unlock.sol [hunk 1]
```diff
@@ -276,11 +276,7 @@ contract Unlock is UnlockInitializable, UnlockOwnable {
     }
 
     // deploy a proxy pointing to impl
-    TransparentUpgradeableProxy proxy = new TransparentUpgradeableProxy(
-      publicLockImpl,
-      proxyAdminAddress,
-      data
-    );
+    LockProxy proxy = new LockProxy(publicLockImpl, proxyAdminAddress, data);
```
Comment (julien51) [inline smart-contracts/contracts/Unlock.sol:279]: Ho wow, we already had the getter/setter?

**Review by julien51 — COMMENTED** [review]
I am confused a bit here. Will it actually change anything on our end?

File: smart-contracts/contracts/utils/LockProxy.sol [hunk 2]
```diff
@@ -0,0 +1,954 @@
+// Sources flattened with hardhat v2.18.3 https://hardhat.org
```
Comment (julien51) [inline smart-contracts/contracts/utils/LockProxy.sol:1]: Is this just the OZ one? or did we add anything custom?
If not, maybe it's better to use the OZ one?

File: smart-contracts/contracts/utils/LockProxy.sol [hunk 3]
```diff
@@ -0,0 +1,954 @@
...
+        require(msg.value == 0);
+    }
+}
+
+
+// File contracts/utils/LockProxy.sol
+
+interface ILockProxy is ITransparentUpgradeableProxy {}
+
+contract LockProxy is TransparentUpgradeableProxy {
```
Comment (julien51) [inline smart-contracts/contracts/utils/LockProxy.sol:948]: yeah it looks like we could just use `ITransparentUpgradeableProxy` directly?

**Comment by clemsos** [comment]
reopening as the approach in #13088 won't work until we maintain a custom proxy

File: smart-contracts/contracts/utils/LockProxy.sol (hunk 3 above)
Comment (clemsos) [reply smart-contracts/contracts/utils/LockProxy.sol:948]: yes we can remove the concept of `LockProxy` and keep only the native (flatten) `TransparentUpgradeableProxy` from Open Zeppelin

**Review by julien51 — APPROVED** [review]

**Pull request merged**